pontoon/base/static/js/translate.js

```diff
@@ -1842,13 +1853,16 @@ var Pontoon = (function (my) {
 
         case "INACTIVE":
           if (!Pontoon.app.advanced && $("#editor").is('.opened')) {
-            $('#cancel').trigger('click', [{inplace: true}]);
+            Pontoon.cancelTranslation();
           }
           break;
 
         case "UPDATE":
-          var entity = Pontoon.entities[message.value.id];
-          Pontoon.updateOnServer(entity, message.value.content, true, true);
+          var entity = Pontoon.entities[message.value.id],
+              translation = message.value.content;
+          Pontoon.updateOnServer(entity, translation, true, true);
+          $('#translation').val(translation);
+          Pontoon.updateCachedTranslation();
```

Comment:
This is a bugfix I noticed while working on this improvement. It caused unsaved changes warning to appear if translation was submitted inplace followed by openning another entity.